cpu/efm32/periph/flashpage.c

```diff
@@ -26,18 +26,17 @@
 
 #include "em_msc.h"
 
-void flashpage_write(int page, void *data)
+void flashpage_write(int page, const void *data)
 {
     assert(page < (int)FLASHPAGE_NUMOF);
 
     uint32_t *page_addr = (uint32_t *)flashpage_addr(page);
-    uint32_t *data_addr = (uint32_t *)data;
```

Comment:
I think this should be kept as `(const uint32_t *)data` as it somehow says alignment constraints, even if its note in the API.

Reply:
I did not repeat but its also present in other files.

Reply:
I don't understand why you want that.

In most cases, the implicit cast is still there anyway. And casting does not enforce any alignment. In this example, without casting, you could have a crash if the alignment is not met. If you use this temporary variable and you cast, you will have the right alignment, but you could write data that you do not expect to write. So I'm not sure which is worst, but I think a crash would be better to catch issues here.

Reply:
I would say it should not be changed in this PR that says to only change to `const`. Fixing this it is a different problem.

I agree that the current state is far from perfect. But random crash are not good either.
Having this line helped me debug a crash in someone else code with a buffer coming from the shell and there was no other mention that for this platform it should be aligned and having this in the code helped me find it.

I would prefer an API fix with a per platform type or alignment definition but it would say it should be another PR.

Reply:
OK to re-add the temp variable but I would prefer keeping implicit cast, which is IMO more aligned with the rest of the code base.

Reply:
I re-checked all the changes and my bad, I generalized this line as if you did the same to the other files.

For the other ones you just removed the cast part, which is true does not help at all. Keeping only the assignement to an `uint32_t *` variable is enough I agree. So you can keep them as you changed now as murdock does not issue warnings.

Here however you also removed the `uint32_t *data_addr` instead of only defining it const.
And from the `MSC_WriteWordI` implementation, it also cast the pointer to `uint32_t` somewhere, so I would prefer keeping this `data_addr` intermediate variable.

Reply:
:D I was answering at the same time.

And yes, re-adding the temp variable here is enough for me.

Reply:
OK, will do!

Reply:
Done